src/Components/Scenes/SearchResults/CheckBoxButton/CheckBoxButton.js

```diff
@@ -21,19 +22,21 @@ const CheckBoxButton = props => {
 
   return (
     <React.Fragment>
-      <Button
-        color="primary"
-        variant="contained"
-        disabled={searchResultChecked.length <= 0}
-        className={classes.button}
-        onClick={() => requestAssetsClick(searchResultChecked)}
-      >
-        {searchResultChecked.length === 1
-          ? `Request ${searchResultChecked.length} Asset`
-          : searchResultChecked.length > 1
-          ? `Request ${searchResultChecked.length} Assets`
-          : 'Request Asset'}
-      </Button>
+      <Link to="/RequestAsset" className={classes.underline}>
+        <Button
+          color="primary"
+          variant="contained"
+          disabled={searchResultChecked.length <= 0}
+          className={classes.button}
+          onClick={() => requestAssetsClick(searchResultChecked)}
+        >
+          {searchResultChecked.length === 1
+            ? `Request ${searchResultChecked.length} Asset`
```

Comment:
Could probably define a const outside of the `return` block that looks at the length and spits out a string that's either "Asset" or "Assets". That could keep the markup more concise

In general, might want to keep this type of logic (and anything more complex) out of the `return` block; usually only markup and very minimal conditionals look best there. If you pre-compute the assets buttons, it can just be a one-liner down here

Reply:
Good Catch. Totally agree. We definitely want to separate logic from the JSX.
Solved

Reply:
I totally agree. We definitely want to separate logic from JSX. good catch.. Remove the logic from JSX Solved